tools/autograd/templates/VariableType.cpp

```diff
@@ -1,520 +1,12 @@
-#include "VariableType.h"
+#include "torch/csrc/autograd/VariableTypeImplementation.h"
```

Comment:
Nit: maybe call this VariableTypeUtils.h? That’s definitely less verbose. We usually use XImpl for the pImpl pattern so it might be confusing.